src/databricks/labs/remorph/reconcile/query_builder/expression_generator.py

```diff
@@ -5,14 +5,14 @@
 
 
 def _apply_func_expr(expr: exp.Expression, expr_func: Callable, **kwargs) -> exp.Expression:
-    level = 0 if isinstance(expr, exp.Column) else 1
+    is_terminal = isinstance(expr, exp.Column)
     new_expr = expr.copy()
     for node in new_expr.dfs():
         if isinstance(node, exp.Column):
             column_name = node.name
             table_name = node.table
             func = expr_func(this=exp.Column(this=column_name, table=table_name), **kwargs)
-            if level == 0:
+            if is_terminal:
```

Comment:
does this function intended to be used only for single column?

Reply:
The whole idea is it should work on nested expression or single column expression